tasks/generate_cert.yml

```diff
@@ -46,10 +71,11 @@
     group: www-data
     recurse: true
 
-- name: Restart nginx
+- name: Restart http server
   service:
-    name: nginx
+    name: "{{ certbot_http_service }}"
     state: restarted
+  when: certbot_http_service != false
```

Comment:
I still have doubts about this approach, this role is perfectly valid without any http service. I wonder if there is a way to isolate restarting http server (or actually anything). This is actually the reason why I don't like to use roles from ansible galaxy, the have hidden and hardcoded dependencies like this one from things I might not even use. 
Here http server is hidden under variable that is null by default (which is better than what I have seen so far) but stil this role is closed for extension and open for modifications but it should be opposite. 
Let's brainstorm this, maybe we can find a better solution.